### PR TITLE
Add SameBoy

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/LIJI32/SameBoy/issues/491
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build io.github.sameboy.SameBoy

**Comment by hfiguiere** [comment]
Probably need a `-Wno-error` flag.

**Comment by Mailaender** [comment]
bot, build io.github.sameboy.SameBoy

**Comment by Mailaender** [comment]
bot, build io.github.sameboy.SameBoy

File: io.github.sameboy.SameBoy.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        tag: v0.5.2
+        commit: 9d993d84e85eeb8dd304e48463d89865795e97ed
+
+  - name: sameboy
+    buildsystem: autotools
+    no-autogen: true
+    build-options:
+      arch:
+        aarch64:
+          cflags: -Wno-unused-command-line-argument
```
Comment (hfiguiere) [inline io.github.sameboy.SameBoy.yml:36]:
```suggestion
          cflags: -Wno-error
```

Just not have warning cause error. Unless you are going to actually fixt them

**Comment by Mailaender** [comment]
bot, build io.github.sameboy.SameBoy

File: io.github.sameboy.SameBoy.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.llvm13
+build-options:
+  append-path: /usr/lib/sdk/llvm13/bin
+  prepend-ld-library-path: /usr/lib/sdk/llvm13/lib
+command: sameboy
+rename-desktop-file: sameboy.desktop
+rename-icon-file: sameboy
```
Comment (hfiguiere) [inline io.github.sameboy.SameBoy.yml:12]: it's 
```suggestion
rename-icon: sameboy
```

there is even a warning about that.

**Comment by hfiguiere** [comment]
looks like cflags ain't gonna work.

Instead with that patch it works:
```patch
--- sameboy/Makefile-orig       2022-08-28 11:13:04.326979227 -0400
+++ sameboy/Makefile    2022-08-28 11:13:15.514826393 -0400
@@ -112,7 +112,7 @@
 endif

 # These must come before the -Wno- flags
-WARNINGS += -Werror -Wall -Wno-unknown-warning -Wno-unknown-warning-option -Wno-missing-braces
+WARNINGS += -Wall -Wno-unknown-warning -Wno-unknown-warning-option -Wno-missing-braces
 WARNINGS += -Wno-nonnull -Wno-unused-result -Wno-strict-aliasing -Wno-multichar -Wno-int-in-bool-context -Wno-format-truncation

 # Only add this flag if the compiler supports it
```

tl;dr just remove -Werror.

**Comment by Mailaender** [comment]
bot, build io.github.sameboy.SameBoy

File: io.github.sameboy.SameBoy.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.llvm13
+build-options:
+  append-path: /usr/lib/sdk/llvm13/bin
+  prepend-ld-library-path: /usr/lib/sdk/llvm13/lib
+command: sameboy
+rename-desktop-file: sameboy.desktop
+rename-icon: sameboy
+finish-args:
+  - --socket=x11
```
Comment (barthalion) [inline io.github.sameboy.SameBoy.yml:14]: It works just fine on Wayland.

File: io.github.sameboy.SameBoy.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      - DATA_DIR=${FLATPAK_DEST}/share/sameboy/
+    make-install-args:
+      - PREFIX=${FLATPAK_DEST}
+      - DATA_DIR=${FLATPAK_DEST}/share/sameboy/
+    post-install:
+      # https://github.com/flatpak/flatpak-xdg-utils/issues/35
+      - install -Dm 644 FreeDesktop/sameboy.desktop ${FLATPAK_DEST}/share/applications/sameboy.desktop
+      - install -Dm 644 io.github.sameboy.SameBoy.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: git
```
Comment (barthalion) [inline io.github.sameboy.SameBoy.yml:43]: It would make sense to configure extra-data-checker if it's not going to be maintained by upstream.

File: io.github.sameboy.SameBoy.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      <image type="source">https://dashboard.snapcraft.io/site_media/appmedia/2020/03/Bildschirmfoto_von_2020-03-16_00-58-39_sAm4YZR.png</image>
+    </screenshot>
+    <screenshot>
+      <image type="source">https://dashboard.snapcraft.io/site_media/appmedia/2020/03/Bildschirmfoto_von_2020-03-16_01-00-07.png</image>
+    </screenshot>
+    <screenshot>
+      <image type="source">https://dashboard.snapcraft.io/site_media/appmedia/2020/03/Bildschirmfoto_von_2020-03-16_01-02-12.png</image>
+    </screenshot>
+  </screenshots>
+  <releases>
```
Comment (barthalion) [inline io.github.sameboy.SameBoy.metainfo.xml:28]: There's no point in including full release history given these versions were never available on Flathub in the first place.

Comment (Mailaender) [reply io.github.sameboy.SameBoy.metainfo.xml:28]: This is automatically generated and submitted upstream.

**Comment by Mailaender** [comment]
bot, build io.github.sameboy.SameBoy

**Comment by WingofaGriffin** [comment]
@Mailaender upsteam has been updated https://github.com/LIJI32/SameBoy/releases/latest
Is there a way to automate the build process to sync with upstreams releases?

**Comment by Mailaender** [comment]
Yes, there is https://github.com/flathub/flatpak-external-data-checker but upstream needs to accept https://github.com/LIJI32/SameBoy/pull/492 otherwise there might be conflicts.
